src/main/java/org/eblocker/crypto/pki/PKI.java

```diff
@@ -244,23 +244,22 @@ public static CertificateAndKey generateSelfSignedCertificateRequest(String name
         return new CertificateAndKey(generateCertificate(builder, keyPair.getPrivate()), keyPair.getPrivate());
     }
 
-    public static X509Certificate generateTLSServerCertificate(X509Certificate request, String orgName, String name, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
-        return doGenerateSignedCertificate(request, orgName, name, notValidAfter, l1ca, KeyPurposeId.id_kp_serverAuth);
+    public static X509Certificate generateTLSServerCertificate(X509Certificate request, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
+        return doGenerateSignedCertificate(request, getStartDate(), notValidAfter, l1ca, KeyPurposeId.id_kp_serverAuth);
     }
 
-    public static X509Certificate generateTLSClientCertificate(X509Certificate request, String orgName, String name, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
-        return doGenerateSignedCertificate(request, orgName, name, notValidAfter, l1ca, KeyPurposeId.id_kp_clientAuth);
+    public static X509Certificate generateTLSClientCertificate(X509Certificate request, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
+        return doGenerateSignedCertificate(request, getStartDate(), notValidAfter, l1ca, KeyPurposeId.id_kp_clientAuth);
     }
 
-    public static X509Certificate generateSignedCertificate(X509Certificate request, String orgName, String name, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
-        return doGenerateSignedCertificate(request, orgName, name, notValidAfter, l1ca);
+    public static X509Certificate generateSignedCertificate(X509Certificate request, Date startDate, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
+        return doGenerateSignedCertificate(request, startDate, notValidAfter, l1ca);
     }
 
-    private static X509Certificate doGenerateSignedCertificate(X509Certificate request, String orgName, String name, Date notValidAfter, CertificateAndKey l1ca, KeyPurposeId...keyPurposeIds) throws CryptoException {
```

Comment:
These two parameters orgName and name are not used in this method. I have therefore deleted them.

Reply:
Many thanks for the tip. I'm always happy to receive tips like that. I have added another commit. In this commit I have also changed the call of static methods via the class variable and not via the instance variable.